### PR TITLE
tests: Fixes tests for Windows (containerd, RunAsUserName)

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind failing-test

/sig windows
/sig testing

/area conformance

**What this PR does / why we need it**:

Since we've added support for ``RunAsUserName``, we can now run some new tests. However, the ``[LinuxOnly]`` tag will have to remain until the ``WindowsRunAsUserName`` feature becomes enabled by default.

Additionally, Containerd supports file mounting on Windows, and some tests will be able to pass on Windows with Containerd instead of Docker.

**Which issue(s) this PR fixes**:

Related: #70189
Related: #68791

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/common/util.go [hunk 1]
```diff
@@ -45,6 +45,13 @@ const (
 	NodeE2E Suite = "node e2e"
 )
 
+var (
+	// non-Administrator Windows user used in tests. This is the Windows equivalent of the Linux non-root UID usage.
+	nonAdminTestUserName = "ContainerUser"
+	// non-root UID used in tests.
+	nonRootTestUserID = int64(1000)
```
Comment (jingxu97) [inline test/e2e/common/util.go]: now we can only set userId as 1000 for linux. For current tests, it is fine. Will it become a concern in the future?

Comment (bclau) [reply test/e2e/common/util.go]: Nothing changed, userid 1000 was also used before this change. Other userids can still be used.

**Comment by msau42** [comment]
/approve
will let @jingxu97 give final review

File: test/e2e/common/util.go [hunk 2]
```diff
@@ -44,6 +44,13 @@ const (
 	NodeE2E Suite = "node e2e"
 )
 
+var (
+	// non-Administrator Windows user used in tests. This is the Windows equivalent of the Linux non-root UID usage.
+	nonAdminTestUserName = "ContainerUser"
```
Comment (jingxu97) [inline test/e2e/common/util.go]: maybe make it as constant, so you don't need to define it multiple places?

File: test/e2e/common/util.go (hunk 1 above)
Comment (jingxu97) [reply test/e2e/common/util.go]: Here it is kind of hardcoded to set user id to 1000. My point is all tests will have to share the exact same value which is defined here.

Comment (bclau) [reply test/e2e/common/util.go]: I don't see the problem.

Comment (PatrickLang) [reply test/e2e/common/util.go]: @jingxu97 -  Before this value was split across many places, and this change defines it once. Are you saying you'd prefer the old way where 1000 was individually hardcoded in many test cases?

**Comment by PatrickLang** [comment]
/hold
I'm double checking with conformance on this re: consolidating on UID1000

**Comment by johnbelamaric** [comment]
LGTM from a conformance point of view

**Comment by PatrickLang** [comment]
/hold cancel
thanks @johnbelamaric and @jingxu97

**Pull request merged**